chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -995,6 +995,31 @@ impl PeerManagerActor {
                 );
                 NetworkResponses::NoResponse
             }
+            NetworkRequests::PartialEncodedStateWitness(chunk_validators, partial_witness) => {
+                for chunk_validator in chunk_validators {
+                    self.state.send_message_to_account(
```

Comment:
This part should only be sent to the one validator which is responsible for forwarding it to the others. I'd suggest adding another field `part_distributor` to ` NetworkRequests::PartialEncodedStateWitness` so that the decision of which validator is responsible can be passed here from the state witness actor.

Reply:
Right, yeah! will change. Thanks! In general, is it a better idea to send one network request with Vec<(validator, partial_witness)> or send multiple network requests with one (validator, partial_witness)

Reply:
I don't see any disadvantages of batching the requests into a `Vec`. It's more efficient and the type is internal anyway (just passed between actix actors) so we can change it freely. Batching indicates more clearly what is happening in case we want to think in the direction of optimizing multicast on the network level in the future.